src/main/java/core/basesyntax/db/DbManagerFactory.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.db;
+
+import core.basesyntax.db.csv.Reader;
+import core.basesyntax.db.csv.Writer;
+import core.basesyntax.model.FruitResultingRow;
+import core.basesyntax.model.FruitTransactionRow;
+
+public class DbManagerFactory {
+    private static DbManager<FruitTransactionRow, FruitResultingRow> dbManager;
+
+    public static void initInstance(Reader<FruitTransactionRow> reader,
+                             Writer<FruitResultingRow> writer) {
+        dbManager = new DbManagerImpl(reader, writer);
```

Comment:
database should not have any connection with Reader or Writer
database should be independent. Higher level (service) can use DB.